### PR TITLE
[FreeBSD][Apple] Implement fallback frequency calculation

### PR DESCRIPTION
**Pull request by er2off**
This PR is a recreation of #232 because seems like @Dr-Noob is now ready to merge it.

### CODE REVIEW

**Comment by Dr-Noob** [comment]
General question, why is this needed? Have you found a x86 CPU which does not get the frequency reported when using FreeBSD and macOS? Which CPU is that?

**Comment by er2off** [comment]
> Which CPU is that?

For me it's AMD Ryzen 5 3500, it reports `Can't read frequency information from cpuid (needed level is 0x00000016, max is 0x00000010).`

**Comment by Dr-Noob** [comment]
> > Which CPU is that?
> 
> For me it's AMD Ryzen 5 3500, it reports `Can't read frequency information from cpuid (needed level is 0x00000016, max is 0x00000010).`

I see, makes sense. Then please have a look at my review comment.

**Comment by er2off** [comment]
> I see, makes sense. Then please have a look at my review comment.

Maybe I don't understood something but referenced commit nor next commits doesn't change anything for x86/amd64.

(kinda offtopic for this issue) BTW cpufetch doesn't work on ARM64 FreeBSD (because nobody ported) and u-boot doesn't expose many information about CPU, some related sysctls for rpi3b are
```
hw.fdt.compatible: raspberrypi,3-model-b brcm,bcm2837
hw.fdt.model: Raspberry Pi 3 Model B Rev 1.2
hw.machine: arm64
hw.model: ARM Cortex-A53 r0p4
hw.ncpu: 4
hw.byteorder: 1234
hw.machine_arch: aarch64
```

**Comment by Dr-Noob** [comment]
> > I see, makes sense. Then please have a look at my review comment.
> 
> Maybe I don't understood something but referenced commit nor next commits doesn't change anything for x86/amd64.
> 

There is no visible change in x86. What changed is the fact that you can now include `src/common/sysctl.h` in `src/x86/cpuid.c` and use `get_sys_info_by_name` instead of `sysctlbyname` directly.

> (kinda offtopic for this issue) BTW cpufetch doesn't work on ARM64 FreeBSD (because nobody ported) and u-boot doesn't expose many information about CPU, some related sysctls for rpi3b are
> 
> ```
> hw.fdt.compatible: raspberrypi,3-model-b brcm,bcm2837
> hw.fdt.model: Raspberry Pi 3 Model B Rev 1.2
> hw.machine: arm64
> hw.model: ARM Cortex-A53 r0p4
> hw.ncpu: 4
> hw.byteorder: 1234
> hw.machine_arch: aarch64
> ```
> 
> (neofetch on this platform simply says hw.model value which doesn't represent real cpu name)

Regarding what you mention about ARM64 in FreeBSD, if you have a proper error message from cpufetch please open an issue and we'll see what can be done to support it :+1: .

**Review by Dr-Noob — CHANGES_REQUESTED** [review]

File: src/x86/cpuid.c [hunk 1]
```diff
@@ -710,7 +713,7 @@ struct topology* get_topology_info(struct cpuInfo* cpu, struct cache* cach, int
   }
 
   switch(cpu->cpu_vendor) {
-    case CPU_VENDOR_INTEL:
+    case CPU_VENDOR_INTEL: {
```
Comment (er2off) [inline src/x86/cpuid.c:716]: This was added to resolve clang compile error

Comment (Dr-Noob) [reply src/x86/cpuid.c:716]: I have fixed this in another way (just pushed into master)

**Comment by Dr-Noob** [comment]
Looks good to me! Please rebase and confirm this works for your hardware. If this is the case, I'll merge this into master shortly.

**Comment by Dr-Noob** [comment]
Okay, ready for merge. Just want to confirm if you are ok with this?

Screenshot_20240818_164228

**Comment by er2off** [comment]
> you are ok with this?

Yes (btw this PR was already squashed)

**Comment by Dr-Noob** [comment]
Merged, thank you for your contribution! :+1:

**Pull request merged**